### DISCUSSION TITLE
[Feature] Expand the notification system with webhooks

### DISCUSSION THREAD

**Discussion by miguelangel-nubla** [immich-app/immich#13181] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Add ability to run a webhook that sends event details in JSON to a custom URL when a photo is added.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
There's already a system in place to send websocket messages for events happening inside Immich (for example when a background job finishes processing). I think it's more likely that we expand this system to cover more events and support sending them out on a webhook, rather than adding webhook support to the notification system specifically.

**Reply by miguelangel-nubla** [reply]
I am fine with that, just following up with https://github.com/immich-app/immich/discussions/1671

**Reply by ghost** [reply]
Webhooks would be an amazing addition to immich

**Reply by ChangsoonKim** [reply]
I agree with you, but it doesn't seem bad that webhook exists for interworking with external systems.

**Comment by etymotic** [comment]
As someone who uses ntfy webhooks would be cool.

For everyone else, though, a native notification through the mobile app is the way to go.

**Comment by vwkd** [comment]
I'd like to propose adding a notification library like Apprise, which allows to outsource notifications for many kinds of notification services including email and webhooks and reduce complexity by not having to reimplement them for each service.
